### PR TITLE
story/BODS-65: Test cases for FareFrame mandatory checks - fareProducts

### PR DESCRIPTION
**Pull request by ndemenezes**

### CODE REVIEW

File: transit_odp/fares_validator/tests/test_mandatory_fareframe_fare_products.py [hunk 1]
```diff
@@ -0,0 +1,971 @@
...
+        xml = frames.format(fare_frame_type_of_frame_ref_not_present)
+
+    netex_xml = etree.fromstring(xml)
+    xpath = "//x:dataObjects/x:CompositeFrame/x:frames/x:FareFrame"
+    fare_frames = netex_xml.xpath(
+        xpath, namespaces={"x": "http://www.netex.org.uk/netex"}
+    )
+
+    response = check_preassigned_fare_products("", fare_frames)
+    print("response", response)
```
Comment (pritish-ranjan01) [inline transit_odp/fares_validator/tests/test_mandatory_fareframe_fare_products.py:185]: removed prints

File: transit_odp/fares_validator/tests/test_mandatory_fareframe_fare_products.py [hunk 2]
```diff
@@ -0,0 +1,971 @@
...
+                xml = frames.format(fare_frame_with_all_children_properties)
+            else:
+                xml = frames.format(fare_frame_without_type_of_fare_product_ref)
+        else:
+            xml = frames.format(fare_frame_type_of_frame_ref_not_valid)
+    else:
+        xml = frames.format(fare_frame_type_of_frame_ref_not_present)
+
+    netex_xml = etree.fromstring(xml)
+    xpath = "//x:dataObjects/x:CompositeFrame/x:frames/x:FareFrame/x:fareProducts/x:PreassignedFareProduct"
```
Comment (pritish-ranjan01) [inline transit_odp/fares_validator/tests/test_mandatory_fareframe_fare_products.py:300]: Please add a common function and pass the xpath and the xml

File: transit_odp/fares_validator/tests/test_mandatory_fareframe_fare_products.py [hunk 3]
```diff
@@ -0,0 +1,971 @@
...
+                    xml = frames.format(fare_frame_without_preassigned_fare_product)
+            else:
+                xml = frames.format(fare_frame_without_fare_products)
+        else:
+            xml = frames.format(fare_frame_type_of_frame_ref_not_valid)
+    else:
+        xml = frames.format(fare_frame_type_of_frame_ref_not_present)
+
+    netex_xml = etree.fromstring(xml)
+    xpath = "//x:dataObjects/x:CompositeFrame/x:frames/x:FareFrame"
```
Comment (pritish-ranjan01) [inline transit_odp/fares_validator/tests/test_mandatory_fareframe_fare_products.py:179]: add a common function and pass xpath and xml

File: transit_odp/fares_validator/tests/test_mandatory_fareframe_fare_products.py [hunk 4]
```diff
@@ -0,0 +1,971 @@
...
+                xml = frames.format(fare_frame_without_type_of_fare_product_ref)
+        else:
+            xml = frames.format(fare_frame_type_of_frame_ref_not_valid)
+    else:
+        xml = frames.format(fare_frame_type_of_frame_ref_not_present)
+
+    netex_xml = etree.fromstring(xml)
+    xpath = "//x:dataObjects/x:CompositeFrame/x:frames/x:FareFrame/x:fareProducts/x:PreassignedFareProduct"
+    preassigned_fare_products = netex_xml.xpath(
+        xpath, namespaces={"x": "http://www.netex.org.uk/netex"}
```
Comment (pritish-ranjan01) [inline transit_odp/fares_validator/tests/test_mandatory_fareframe_fare_products.py:302]: add a constant for the namespace

Comment (ndemenezes) [reply transit_odp/fares_validator/tests/test_mandatory_fareframe_fare_products.py]: These changes have been applied

**Review by surabhika — APPROVED** [review]

**Pull request merged**